### ISSUE TITLE
Cmd line reader fails on paths with trailing slash

### ISSUE THREAD

**Issue by Antonius-S** [eslint/eslint#12505]
**Tell us about your environment**

Environment Info:

Windows
Node version: v10.15.1
npm version: v6.12.0
Local ESLint version: Not found
Global ESLint version: v6.6.0 (Currently used)

**What parser (default, Babel-ESLint, etc.) are you using?**
unrelated

**Please show your full configuration:**
unrelated

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
>eslint "--resolve-plugins-relative-to=.\" test-eslint.js
```

**What did you expect to happen?**
Normal run

**What actually happened? Please include the actual, raw output from ESLint.**
Help text printed as if parameters are incorrect

**Are you willing to submit a pull request to fix this bug?**
no

**In other words**: cmd line arguments reader dislikes quoted paths ending with slash. Unquoted paths ending with slash are OK, only two factors  together cause reader to fail.

```bash
>eslint "--resolve-plugins-relative-to=." test-eslint.js - OK
>eslint --resolve-plugins-relative-to=.\ test-eslint.js - OK
```
Dot path is used here for shortness & sample only, the same with any other meaningful path

**Comment by mysticatea** [comment]
Thank you for your report.

This is the behavior of Node.js and `cmd.exe`. The backslash `\` seems to escape the quote `"`. If you used other shells (e.g., PowerShell), it works fine.

```
> node -p "process.argv" -- "--a=foo\" bar
[
  'C:\\Program Files (x86)\\Nodist\\v-x64\\12.12.0\\node.exe',
  '--a=foo" bar'
]
```

Like the above, Node.js receives "`--a=foo" bar`" as one argument. Because we cannot know the existence of the backslash and the quotes, I think that what we can do on the ESLint side is nothing.

**Comment by Antonius-S** [comment]
Hmm, very surprising behavior. Okay, sorry for kinda false report though maybe it could help someone with similar issue. For me the best solution is to add one more slash
```
eslint "--resolve-plugins-relative-to=.\\" test-eslint.js
```
